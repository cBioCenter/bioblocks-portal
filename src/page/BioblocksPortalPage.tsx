import { EMPTY_FUNCTION, fetchDataset } from 'bioblocks-viz';
import * as React from 'react';
import { connect } from 'react-redux';
import { Route, RouteComponentProps, Switch } from 'react-router-dom';
import { bindActionCreators, Dispatch } from 'redux';
import { Container } from 'semantic-ui-react';

import { ConnectedRouter } from 'connected-react-router';
import { SiteHeader } from '~bioblocks-portal~/container';
import { IDataset, IEveResponse, IVignette, IVisualization } from '~bioblocks-portal~/data';
import {
  DatasetsPage,
  DynamicsPage,
  LandingPage,
  VignettesPage,
  VisualizationsPage,
  VizOverviewPage,
} from '~bioblocks-portal~/page';
import { history, IPortalReducerState } from '~bioblocks-portal~/reducer';
import { selectDatasets, selectVignettes, selectVisualizations } from '~bioblocks-portal~/selector';

export interface IBioblocksPortalPageProps {
  datasets: IDataset[];
  vignettes: IVignette[];
  visualizations: IVisualization[];
  dispatchDatasetsFetch(dataset: string, fetchFn: () => Promise<IDataset[]>): void;
  dispatchVignettesFetch(dataset: string, fetchFn: () => Promise<IVignette[]>): void;
  dispatchVisualizationsFetch(dataset: string, fetchFn: () => Promise<IVisualization[]>): void;
}

export class UnconnectedBioblocksPortalPage extends React.Component<IBioblocksPortalPageProps> {
  public static defaultProps = {
    datasets: new Array<IDataset>(),
    dispatchDatasetsFetch: EMPTY_FUNCTION,
    dispatchVignettesFetch: EMPTY_FUNCTION,
    dispatchVisualizationsFetch: EMPTY_FUNCTION,
    vignettes: new Array<IVignette>(),
    visualizations: new Array<IVisualization>(),
  };

  constructor(props: IBioblocksPortalPageProps) {
    super(props);
  }

  public async componentDidMount() {
    const { dispatchDatasetsFetch, dispatchVignettesFetch, dispatchVisualizationsFetch } = this.props;

    dispatchDatasetsFetch('datasets', this.onDatasetFetch);

    dispatchVignettesFetch('vignettes', this.onVignetteFetch);

    dispatchVisualizationsFetch('visualizations', this.onVisualizationFetch);
  }

  public render() {
    return (
      <ConnectedRouter history={history}>
        <Container id={'BioblocksPortalPage'} fluid={true}>
          <SiteHeader {...this.props} />

          <Switch>
            <Route exact={true} strict={true} path={'/datasets'} render={this.renderDatasetsPage} />
            <Route exact={true} strict={true} path={'/visualizations'} render={this.renderVisualizationsPage} />
            <Route exact={true} strict={true} path={'/visualizations/'} render={this.renderVizOverviewPage} />
            <Route path={'/dynamics'} render={this.renderDynamicsPage} />
            <Route path={'/vignettes'} render={this.renderVignettesPage} />
            <Route exact={true} path={'/'} render={this.renderLandingPage} />
          </Switch>
        </Container>
      </ConnectedRouter>
    );
  }

  protected onDatasetFetch = async () => {
    const datasetFetchResult = await fetch(`${process.env.API_URL}/dataset?embedded={"analyses": 1}`);
    if (!datasetFetchResult.ok) {
      return [];
    } else {
      const response = (await datasetFetchResult.json()) as IEveResponse<IDataset>;

      return response._items.sort((a, b) => a.name.localeCompare(b.name));
    }
  };

  protected onVignetteFetch = async () => {
    const vignetteFetchResult = await fetch(`${process.env.API_URL}/vignette`);
    if (!vignetteFetchResult.ok) {
      return [];
    } else {
      const response = (await vignetteFetchResult.json()) as IEveResponse<IVignette>;

      return response._items.sort((a, b) => a.name.localeCompare(b.name));
    }
  };

  protected onVisualizationFetch = async () => {
    const visualizationFetchResult = await fetch(`${process.env.API_URL}/visualization`);
    if (!visualizationFetchResult.ok) {
      return [];
    } else {
      const response = (await visualizationFetchResult.json()) as IEveResponse<IVisualization>;

      return response._items.sort((a, b) => a.name.localeCompare(b.name));
    }
  };

  protected renderDynamicsPage = () => {
    return <DynamicsPage />;
  };

  protected renderLandingPage = (props: RouteComponentProps) => {
    const { vignettes, visualizations } = this.props;

    const featuredVignettes = new Array<IVignette>();
    if (vignettes) {
      const tabulaVignette = vignettes.find(vignette => vignette._id === 'bbdecade-0000-0000-0001-a1234567890b');
      const springVignette = vignettes.find(vignette => vignette._id === 'bbdecade-0000-0000-0002-a1234567890b');

      if (springVignette) {
        featuredVignettes.push(springVignette);
      }
      if (tabulaVignette) {
        featuredVignettes.push(tabulaVignette);
      }
    }

    const featuredVisualizations = new Array<IVisualization>();
    if (visualizations) {
      const springViz = visualizations.find(viz => viz.name === 'SPRING');
      const umapViz = visualizations.find(viz => viz.name === 'UMAP');

      if (springViz) {
        featuredVisualizations.push(springViz);
      }
      if (umapViz) {
        featuredVisualizations.push(umapViz);
      }
    }

    return (
      <LandingPage {...props} featuredVignettes={featuredVignettes} featuredVisualizations={featuredVisualizations} />
    );
  };

  protected renderDatasetsPage = (props: RouteComponentProps) => {
    const { datasets, visualizations } = this.props;

    return <DatasetsPage datasets={datasets} visualizations={visualizations} {...props} />;
  };

  protected renderVignettesPage = (props: RouteComponentProps) => {
    const { vignettes } = this.props;

    return <VignettesPage {...props} vignettes={vignettes} />;
  };

  protected renderVisualizationsPage = (props: RouteComponentProps) => {
    const { visualizations } = this.props;

    return <VisualizationsPage {...props} visualizations={visualizations} />;
  };

  protected renderVizOverviewPage = (props: RouteComponentProps) => {
    const { vignettes, visualizations } = this.props;

    return <VizOverviewPage {...props} vignettes={vignettes} visualizations={visualizations} />;
  };
}

const mapStateToProps = (state: IPortalReducerState) => ({
  datasets: selectDatasets(state),
  vignettes: selectVignettes(state),
  visualizations: selectVisualizations(state),
});

const mapDispatchToProps = (dispatch: Dispatch) =>
  bindActionCreators(
    {
      dispatchDatasetsFetch: fetchDataset,
      dispatchVignettesFetch: fetchDataset,
      dispatchVisualizationsFetch: fetchDataset,
    },
    dispatch,
  );

export const BioblocksPortalPage = connect(
  mapStateToProps,
  mapDispatchToProps,
)(UnconnectedBioblocksPortalPage);
