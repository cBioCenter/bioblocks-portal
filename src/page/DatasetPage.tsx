import * as React from 'react';
import { Grid, Message } from 'semantic-ui-react';

import { AnatomogramContainer, ISpringGraphData, SPECIES_TYPE, SpringContainer, TensorTContainer } from 'bioblocks-viz';
import { connect } from 'react-redux';
import { IVignette, IVisualization } from '~bioblocks-portal~/data';
import { IPortalReducerState } from '~bioblocks-portal~/reducer';
import { selectVignettes, selectVisualizations } from '~bioblocks-portal~/selector';

export interface IDatasetPageProps {
  pathname: string;
  search: string;
  vignettes: IVignette[];
  visualizations: IVisualization[];
  dispatchSpringFetch(fetchFn: () => Promise<ISpringGraphData>, namespace?: string): void;
  setSpecies(species: SPECIES_TYPE): void;
}

export interface IDatasetPageState {
  datasetVisualizations: IVisualization[];
  datasetLocation: string;
}

export class UnconnectedDatasetPage extends React.Component<IDatasetPageProps, IDatasetPageState> {
  public static defaultProps = {
    dispatchSpringFetch: () => {
      return;
    },
    setSpecies: () => {
      return;
    },
  };

  constructor(props: IDatasetPageProps) {
    super(props);
    this.state = {
      datasetLocation: 'hpc_sf2/full',
      datasetVisualizations: [],
    };
  }

  public async componentDidMount() {
    const { search, setSpecies } = this.props;
    const { datasetLocation } = this.state;
    if (search) {
      this.setupSearchParameters(search);
    }
    setSpecies(datasetLocation.includes('hpc') ? 'homo_sapiens' : 'mus_musculus');
  }

  public componentDidUpdate(prevProps: IDatasetPageProps) {
    const { search, setSpecies, vignettes, visualizations } = this.props;
    const { datasetLocation } = this.state;
    if (
      (search && search !== prevProps.search) ||
      vignettes !== prevProps.vignettes ||
      visualizations !== prevProps.visualizations
    ) {
      this.setupSearchParameters(search);
    }
    setSpecies(datasetLocation.includes('hpc') ? 'homo_sapiens' : 'mus_musculus');
  }

  public render() {
    const { datasetVisualizations, datasetLocation } = this.state;

    return (
      <div style={{ padding: '20px' }}>
        <Grid centered={true} stackable={true} stretched={false} padded={true} columns={2}>
          {datasetLocation.length >= 1 &&
            datasetVisualizations.map((visualization, index) => (
              <Grid.Column key={`dataset-visualization-${index}`} style={{ width: 'auto' }}>
                {this.renderVisualization(visualization, datasetLocation)}
              </Grid.Column>
            ))}
        </Grid>
      </div>
    );
  }

  protected setupSearchParameters(query: string) {
    const { vignettes, visualizations } = this.props;

    let datasetLocation = this.state.datasetLocation;
    const datasetVisualizations = new Array<IVisualization>();
    if (vignettes) {
      const vignetteParams = new URLSearchParams(query);
      // tslint:disable-next-line:no-backbone-get-set-outside-model
      const datasetId = vignetteParams.get('id');

      datasetLocation = datasetId ? datasetId : datasetLocation;
      const vizIds = vignetteParams.getAll('viz');
      vizIds.forEach(vizId => {
        const viz = visualizations.find(aViz => vizId === aViz._id);
        if (viz) {
          datasetVisualizations.push(viz);
        }
      });

      this.setState({
        datasetLocation,
        datasetVisualizations,
      });
    }
  }

  protected renderVisualization(viz: IVisualization, datasetLocation: string) {
    const isFullPage = this.state.datasetVisualizations.length === 1;
    const iconSrc = `${process.env.API_URL}${viz.icon}`;
    console.log(datasetLocation);
    switch (viz.name.toLocaleLowerCase()) {
      case 'spring':
        return (
          <SpringContainer
            datasetLocation={datasetLocation}
            datasetsURI={`${process.env.API_URL}/datasets`}
            isFullPage={isFullPage}
            springSrc={`${process.env.API_URL}/${viz.location}`}
            iconSrc={iconSrc}
          />
        );
      case 'tfjs-tsne':
        return (
          <TensorTContainer
            datasetLocation={`${process.env.API_URL}/datasets/${datasetLocation}`}
            iconSrc={iconSrc}
            isFullPage={isFullPage}
          />
        );
      case 'anatomogram':
        return <AnatomogramContainer iconSrc={iconSrc} />;
      default:
        return <Message error={true}>{`Currently unsupported visualization '${viz}'`}</Message>;
    }
  }
}

const mapStateToProps = (state: IPortalReducerState) => ({
  pathname: state.router.location.pathname,
  search: state.router.location.search,
  vignettes: selectVignettes(state),
  visualizations: selectVisualizations(state),
});

export const DatasetPage = connect(mapStateToProps)(UnconnectedDatasetPage);
