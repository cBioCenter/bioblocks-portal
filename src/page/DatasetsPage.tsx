// ~bioblocks-portal~
// Datasets Page
// Page for displaying datasets available on bioblocks-server.
// ~bioblocks-portal~

import * as React from 'react';
import { RouteComponentProps } from 'react-router';
import { Button, Divider, Grid, Header } from 'semantic-ui-react';

import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { IAnalysis, IDataset, IVisualization } from '~bioblocks-portal~/data';

export interface IDatasetsPageProps extends Partial<RouteComponentProps> {
  datasets: IDataset[];
  visualizations: IVisualization[];
}

export class UnconnectedDatasetsPage extends React.Component<IDatasetsPageProps, any> {
  public static defaultProps = {
    datasets: [],
    visualizations: [],
  };

  constructor(props: IDatasetsPageProps) {
    super(props);
  }

  public render() {
    const { datasets } = this.props;

    return (
      <Grid centered={true} padded={true} relaxed={true}>
        {datasets
          .sort((a, b) => b.analyses.length - a.analyses.length)
          .map((dataset, index) => (
            <React.Fragment key={`vignette-${index}`}>{this.renderSingleDatasetEntry(dataset)}</React.Fragment>
          ))}
      </Grid>
    );
  }

  protected getValidVizIds = (analyses: IAnalysis[]) => {
    let result = '';

    for (const analysis of analyses) {
      if (analysis.processType === 'SPRING') {
        result += '&viz=bbfacade-0000-0000-0001-a1234567890b';
        result += '&viz=bbfacade-0000-0000-0004-a1234567890b';
      } else if (analysis.processType === 'TSNE') {
        // result += '&viz=bbfacade-0000-0000-0003-a1234567890b';
      } else if (analysis.processType === 'UMAP') {
        result += '&viz=bbfacade-0000-0000-0004-a1234567890b';
      } else {
        console.log(`Process ${analysis._id} has unhandled process of type '${analysis.processType}'.`);
      }
    }

    return result;
  };

  protected renderSingleDatasetEntry(dataset: IDataset) {
    return (
      <>
        <Grid.Row columns={3}>{this.renderSingleDatasetEntryHeader(dataset)}</Grid.Row>
        <Divider section={true} />
      </>
    );
  }

  protected renderDatasetAnalysisLink(analysis: IAnalysis, datasetId: string) {
    const linkTo = `/dynamics?id=${datasetId}&analysis=${analysis._id}${this.getValidVizIds([analysis])}`;

    return (
      <Button basic={true} key={`launch-button-for-${analysis._id}`}>
        <Link to={linkTo}>{`Launch '${analysis.name}'`}</Link>
      </Button>
    );
  }

  protected renderSingleDatasetEntryHeader(dataset: IDataset) {
    return (
      <Grid.Column textAlign={'left'}>
        <Header>{dataset.name}</Header>

        <Grid.Row>{dataset.analyses.map(analysis => this.renderDatasetAnalysisLink(analysis, dataset._id))}</Grid.Row>
      </Grid.Column>
    );
  }
}

export const DatasetsPage = connect(undefined)(UnconnectedDatasetsPage);
