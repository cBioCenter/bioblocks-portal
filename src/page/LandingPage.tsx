import * as React from "react";
import { RouteComponentProps } from "react-router";
import { Link } from "react-router-dom";
import {
  Button,
  Container,
  Divider,
  Grid,
  Header,
  Segment
} from "semantic-ui-react";

import { IVignette, IVisualization } from "~bioblocks-portal~/data";

export interface ILandingPageProps extends Partial<RouteComponentProps> {
  featuredVignettes: IVignette[];
  featuredVisualizations: IVisualization[];
}

export class LandingPage extends React.Component<ILandingPageProps, any> {
  public static defaultProps = {
    featuredVignettes: [],
    featuredVisualizations: []
  };

  constructor(props: ILandingPageProps) {
    super(props);
  }

  public render() {
    return (
      <Container fluid={false}>
        <Segment basic={true} padded={"very"}>
          <Grid centered={true} padded={true} relaxed={true}>
            {this.renderFeaturedVignettes()}
            {this.renderFeaturedVisualizations()}
            {this.renderFeaturedDatasets()}
          </Grid>
        </Segment>
      </Container>
    );
  }

  protected renderSingleVisualization(viz: IVisualization) {
    return (
      <Grid.Row columns={3}>
        <Grid.Column width={2}>
          <img
            src={`${process.env.API_URL}${viz.icon}`}
            style={{ height: "90px", width: "90px" }}
            alt={`viz ${viz.name} icon`}
          />
        </Grid.Column>
        <Grid.Column textAlign={"left"} width={8}>
          <Header>{viz.name}</Header>
          {viz.summary}
          <p>
            <span style={{ fontWeight: "bold" }}>
              {viz.isOriginal ? "Original authors" : "Authors"}:{" "}
            </span>
            {viz.authors.length === 2
              ? `${viz.authors[0]} and ${viz.authors[1]}`
              : viz.authors.join(", ")}
          </p>
          <br />
        </Grid.Column>
        <Grid.Column width={2}>
          <Grid.Row>
            <Button basic={true}>
              <Link
                to={{ pathname: "/visualizations/", search: `?id=${viz._id}` }}
              >
                details
              </Link>
            </Button>
          </Grid.Row>
          <Grid.Row>
            <Button basic={true}>
              <Link
                to={{
                  pathname: "/dataset",
                  search: `?id=${viz.exampleDataset}&viz=${viz._id}`
                }}
              >
                launch
              </Link>
            </Button>
          </Grid.Row>
        </Grid.Column>
      </Grid.Row>
    );
  }

  protected renderFeaturedDatasets() {
    return (
      <>
        <Grid.Row centered={true}>
          <Grid.Column width={12}>
            <Header floated={"left"}>Featured Datasets</Header>
            <Divider section={true} />
          </Grid.Column>
        </Grid.Row>
        <Grid.Row columns={2} divided={true}>
          {this.renderHCADatasets()}
          {this.renderUserSharedDatasets()}
        </Grid.Row>
      </>
    );
  }

  protected renderFeaturedVignettes() {
    const { featuredVignettes } = this.props;

    return (
      <>
        <Grid.Row centered={false}>
          <Grid.Column width={12}>
            <Header floated={"left"}>Featured Vignettes</Header>
            <Divider section={true} />
          </Grid.Column>
        </Grid.Row>
        {featuredVignettes[0] &&
          this.renderSingleFeaturedVignette(featuredVignettes[0])}
        {featuredVignettes[1] &&
          this.renderSingleFeaturedVignette(featuredVignettes[1])}
        <Grid.Row centered={false}>
          <Grid.Column width={12}>
            <Link style={{ color: "blue", float: "right" }} to={"vignettes"}>
              more vignettes...
            </Link>
          </Grid.Column>
        </Grid.Row>
      </>
    );
  }

  protected renderFeaturedVisualizations() {
    const { featuredVisualizations } = this.props;

    return (
      <>
        <Grid.Row centered={false}>
          <Grid.Column width={12}>
            <Header floated={"left"}>Featured Visualizations</Header>
            <Divider section={true} />
          </Grid.Column>
        </Grid.Row>
        {featuredVisualizations[0] &&
          this.renderSingleVisualization(featuredVisualizations[0])}
        {featuredVisualizations[1] &&
          this.renderSingleVisualization(featuredVisualizations[1])}
        <Grid.Row centered={false}>
          <Grid.Column width={12}>
            <Link
              style={{ color: "blue", float: "right" }}
              to={"visualizations"}
            >
              more visualizations...
            </Link>
          </Grid.Column>
        </Grid.Row>
      </>
    );
  }

  protected renderSingleFeaturedVignette(vignette: IVignette) {
    return (
      <Grid.Row columns={3}>
        <Grid.Column width={2}>
          <img
            src={`${process.env.API_URL}${vignette.icon}`}
            style={{ height: "90px", width: "90px" }}
            alt={`vignettes ${vignette.name} icon`}
          />
        </Grid.Column>
        <Grid.Column textAlign={"left"} width={8}>
          <Header>{vignette.name}</Header>
          <p>
            <span style={{ fontWeight: "bold" }}>Description: </span>
            {vignette.summary}
          </p>
          <p>
            <span style={{ fontWeight: "bold" }}>Analysis authors: </span>
            {vignette.authors.length === 2
              ? `${vignette.authors[0]} and ${vignette.authors[1]}`
              : vignette.authors.join(", ")}
          </p>
          <br />
        </Grid.Column>
        <Grid.Column width={2}>
          <Button basic={true}>
            <Link
              to={`/dataset?id=${vignette.dataset}${vignette.visualizations
                .map(viz => `&viz=${viz}`)
                .join("")}`}
            >
              launch
            </Link>
          </Button>
        </Grid.Column>
      </Grid.Row>
    );
  }

  protected renderHCADatasets() {
    return (
      <Grid.Column width={3}>
        <Grid.Row>
          <img
            src={"assets/icons/hca-logo.png"}
            style={{ width: "150px", height: "250px" }}
            alt={"human cell atlas logo"}
          />
          <Grid.Row>
            <a href={"https://preview.data.humancellatlas.org/"}>
              Search HCA datasets ...
            </a>
          </Grid.Row>
        </Grid.Row>
      </Grid.Column>
    );
  }

  /*
  protected renderSingleUserSharedDatasets(dataset: IDatasetInfo) {
    return (
      <Grid.Row columns={2}>
        <Grid.Column floated={'left'}>
          <Header>{dataset.name}</Header>
          <p>
            {dataset.summary}
            <br />
            <span style={{ fontWeight: 'bold' }}>Authors: </span>
            {dataset.authors.join(', ')}
          </p>
        </Grid.Column>
        <Grid.Column floated={'right'}>
          <Button basic={true} floated={'right'}>
            <Link to={dataset.links.detail}>details</Link>
          </Button>
          <Button basic={true} floated={'right'}>
            <Link to={dataset.links.analysis}>analyze</Link>
          </Button>
        </Grid.Column>
      </Grid.Row>
    );
  }
  */

  protected renderUserSharedDatasets() {
    return (
      <Grid.Column stretched={true} textAlign={"left"} width={8}>
        <Grid.Row textAlign={"center"}>
          <Header as={"h2"} textAlign={"center"}>
            User shared
          </Header>
        </Grid.Row>
        <br />
        <Grid.Row textAlign={"center"}>
          <Header as={"h3"} textAlign={"center"}>
            Coming soon!
          </Header>
        </Grid.Row>
      </Grid.Column>
    );
  }
}
