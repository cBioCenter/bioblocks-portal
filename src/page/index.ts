export * from './BioblocksPortalPage';
export * from './VisualizationsPage';
export * from './DatasetPage';
export * from './LandingPage';
export * from './VignettesPage';
export * from './VizOverviewPage';
