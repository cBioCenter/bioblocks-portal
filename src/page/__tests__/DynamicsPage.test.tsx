import { mount, shallow } from 'enzyme';
import * as React from 'react';

import { ConnectedRouter } from 'connected-react-router';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { DynamicsPage, UnconnectedDynamicsPage } from '~bioblocks-portal~/page';
import { configureStore } from '~bioblocks-portal~/reducer';
import { originalViz, testVignettes, testVisualizations } from '~bioblocks-portal~/test';

describe('DynamicsPage', () => {
  const visualizations = ['anatomogram', 'spring', 'tfjs-tsne', 'umap'];

  it('Should match existing snapshot when no props are provided.', () => {
    const wrapper = shallow(<UnconnectedDynamicsPage />);
    expect(wrapper).toMatchSnapshot();
  });

  // tslint:disable-next-line: mocha-no-side-effect-code
  visualizations.forEach(viz => {
    it(`Should match existing snapshot for initial visualization '${viz}'`, done => {
      const testVisualization = {
        ...originalViz,
        name: viz,
      };
      const wrapper = shallow(
        <UnconnectedDynamicsPage
          search={`/dynamics?id=psycho-jukebox-vignette-id&viz=${originalViz._id}`}
          vignettes={testVignettes}
          visualizations={[testVisualization]}
        />,
      );
      wrapper.setProps({
        dataset: {
          _id: 'sample-dataset-id',
          analyses: [],
          authors: [],
          derivedFrom: [],
          matrixLocation: 'the-church-of-rock-and-roll',
          name: 'sample-dataset',
        },
      });
      wrapper.instance().forceUpdate(() => {
        expect(wrapper).toMatchSnapshot();
        done();
      });
    });
  });

  it('Should match existing snapshot when hooked up to a Redux store.', () => {
    const store = configureStore();
    const history = createMemoryHistory();
    const wrapper = mount(
      <Provider store={store}>
        <ConnectedRouter history={history}>
          <DynamicsPage vignettes={testVignettes} visualizations={testVisualizations} />
        </ConnectedRouter>
      </Provider>,
    );
    history.push('/dynamics?id=psycho-jukebox-vignette-id');
    wrapper.update();
    expect(wrapper.find(DynamicsPage)).toMatchSnapshot();
  });

  it('Should match existing snapshot when given an analysis.', () => {
    const store = configureStore();
    const history = createMemoryHistory();
    const wrapper = mount(
      <Provider store={store}>
        <ConnectedRouter history={history}>
          <DynamicsPage vignettes={testVignettes} visualizations={testVisualizations} />
        </ConnectedRouter>
      </Provider>,
    );
    history.push('/dynamics?analysis=aaaaaaaa-0000-0000-0001-a1234567890b');
    wrapper.update();
    expect(wrapper.find(DynamicsPage)).toMatchSnapshot();
  });

  it('Should match existing snapshot when given an Anatomogram analysis.', () => {
    const store = configureStore();
    const history = createMemoryHistory();
    const wrapper = mount(
      <UnconnectedDynamicsPage
        dataset={{
          _id: 'sample-dataset-id',
          analyses: [],
          authors: [],
          derivedFrom: [],
          matrixLocation: 'the-church-of-rock-and-roll',
          name: 'sample-dataset',
          species: 'mus_musculus' as const,
        }}
        search={`/dynamics?id=psycho-jukebox-vignette-id&viz=${originalViz._id}`}
        vignettes={testVignettes}
        visualizations={[{ ...originalViz, name: 'anatomogram' }]}
      />,
    );
    expect(wrapper.find(DynamicsPage)).toMatchSnapshot();
  });
});
