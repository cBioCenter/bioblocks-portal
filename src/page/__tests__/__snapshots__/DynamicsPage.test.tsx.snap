// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicsPage Should match existing snapshot for initial visualization 'anatomogram' 1`] = `
<div
  style={
    Object {
      "padding": "20px",
    }
  }
>
  <Grid
    centered={true}
    columns={2}
    padded={true}
    stackable={true}
    stretched={false}
  >
    <GridColumn
      key="dataset-visualization-0"
      style={
        Object {
          "width": "auto",
        }
      }
    >
      <Component
        iconSrc="undefinedundefined"
        species="homo_sapiens"
      />
    </GridColumn>
  </Grid>
</div>
`;

exports[`DynamicsPage Should match existing snapshot for initial visualization 'spring' 1`] = `
<div
  style={
    Object {
      "padding": "20px",
    }
  }
>
  <Grid
    centered={true}
    columns={2}
    padded={true}
    stackable={true}
    stretched={false}
  >
    <GridColumn
      key="dataset-visualization-0"
      style={
        Object {
          "width": "auto",
        }
      }
    >
      <Component
        datasetLocation="datasets/hpc_sf2/full/analyses//sample-dataset"
        datasetsURI="undefined/datasets"
        iconSrc="undefinedundefined"
        isFullPage={true}
        springSrc="undefined/undefined"
      />
    </GridColumn>
  </Grid>
</div>
`;

exports[`DynamicsPage Should match existing snapshot for initial visualization 'tfjs-tsne' 1`] = `
<div
  style={
    Object {
      "padding": "20px",
    }
  }
>
  <Grid
    centered={true}
    columns={2}
    padded={true}
    stackable={true}
    stretched={false}
  >
    <GridColumn
      key="dataset-visualization-0"
      style={
        Object {
          "width": "auto",
        }
      }
    >
      <Component
        datasetLocation="undefined/datasets/hpc_sf2/full/analyses/"
        iconSrc="undefinedundefined"
      />
    </GridColumn>
  </Grid>
</div>
`;

exports[`DynamicsPage Should match existing snapshot for initial visualization 'umap' 1`] = `
<div
  style={
    Object {
      "padding": "20px",
    }
  }
>
  <Grid
    centered={true}
    columns={2}
    padded={true}
    stackable={true}
    stretched={false}
  >
    <GridColumn
      key="dataset-visualization-0"
      style={
        Object {
          "width": "auto",
        }
      }
    >
      <GridColumn>
        Loading...
      </GridColumn>
    </GridColumn>
  </Grid>
</div>
`;

exports[`DynamicsPage Should match existing snapshot when given an Anatomogram analysis. 1`] = `null`;

exports[`DynamicsPage Should match existing snapshot when given an analysis. 1`] = `
<Connect(UnconnectedDynamicsPage)
  vignettes={
    Array [
      Object {
        "_id": "bbdecade-0000-0000-0001-a1234567890b",
        "authors": Array [],
        "dataset": "",
        "icon": "",
        "name": "",
        "summary": "Rise Against Viz",
        "visualizations": Array [
          "bbfacade-0000-0000-0001-a1234567890b",
        ],
      },
    ]
  }
  visualizations={
    Array [
      Object {
        "_id": "bbfacade-0000-0000-0001-a1234567890b",
        "authors": Array [
          "Franz Ferdinand",
        ],
        "citations": Array [
          Object {
            "fullCitation": "A full citation",
            "link": "A link",
          },
        ],
        "compatibleData": Array [],
        "exampleDataset": "bbdeeded-0000-0000-0001-a1234567890b",
        "isOriginal": true,
        "labels": Array [],
        "name": "",
        "repo": Object {
          "lastUpdate": "03/13/19",
          "link": "the-dark-of-the-matinée",
          "version": "",
        },
        "summary": "",
        "version": "",
      },
      Object {
        "_id": "bbfacade-0000-0000-0002-a1234567890b",
        "authors": Array [
          "Eminem",
        ],
        "citations": Array [
          Object {
            "fullCitation": "A full citation",
            "link": "A link",
          },
        ],
        "compatibleData": Array [],
        "exampleDataset": "bbdeeded-0000-0000-0002-a1234567890b",
        "isOriginal": false,
        "labels": Array [],
        "name": "",
        "repo": Object {
          "lastUpdate": "03/13/19",
          "link": "The Ringer",
          "version": "",
        },
        "summary": "",
        "version": "",
      },
    ]
  }
>
  <UnconnectedDynamicsPage
    dataset={null}
    dispatchDatasetFetch={[Function]}
    pathname="/dynamics"
    search="?analysis=aaaaaaaa-0000-0000-0001-a1234567890b"
    vignettes={Array []}
    visualizations={Array []}
  >
    <div
      style={
        Object {
          "padding": "20px",
        }
      }
    >
      <Grid
        centered={true}
        columns={2}
        padded={true}
        stackable={true}
        stretched={false}
      >
        <div
          className="ui centered stackable padded two column grid"
        >
          <GridColumn>
            <div
              className="column"
            >
              No visualizations!
            </div>
          </GridColumn>
        </div>
      </Grid>
    </div>
  </UnconnectedDynamicsPage>
</Connect(UnconnectedDynamicsPage)>
`;

exports[`DynamicsPage Should match existing snapshot when hooked up to a Redux store. 1`] = `
<Connect(UnconnectedDynamicsPage)
  vignettes={
    Array [
      Object {
        "_id": "bbdecade-0000-0000-0001-a1234567890b",
        "authors": Array [],
        "dataset": "",
        "icon": "",
        "name": "",
        "summary": "Rise Against Viz",
        "visualizations": Array [
          "bbfacade-0000-0000-0001-a1234567890b",
        ],
      },
    ]
  }
  visualizations={
    Array [
      Object {
        "_id": "bbfacade-0000-0000-0001-a1234567890b",
        "authors": Array [
          "Franz Ferdinand",
        ],
        "citations": Array [
          Object {
            "fullCitation": "A full citation",
            "link": "A link",
          },
        ],
        "compatibleData": Array [],
        "exampleDataset": "bbdeeded-0000-0000-0001-a1234567890b",
        "isOriginal": true,
        "labels": Array [],
        "name": "",
        "repo": Object {
          "lastUpdate": "03/13/19",
          "link": "the-dark-of-the-matinée",
          "version": "",
        },
        "summary": "",
        "version": "",
      },
      Object {
        "_id": "bbfacade-0000-0000-0002-a1234567890b",
        "authors": Array [
          "Eminem",
        ],
        "citations": Array [
          Object {
            "fullCitation": "A full citation",
            "link": "A link",
          },
        ],
        "compatibleData": Array [],
        "exampleDataset": "bbdeeded-0000-0000-0002-a1234567890b",
        "isOriginal": false,
        "labels": Array [],
        "name": "",
        "repo": Object {
          "lastUpdate": "03/13/19",
          "link": "The Ringer",
          "version": "",
        },
        "summary": "",
        "version": "",
      },
    ]
  }
>
  <UnconnectedDynamicsPage
    dataset={null}
    dispatchDatasetFetch={[Function]}
    pathname="/dynamics"
    search="?id=psycho-jukebox-vignette-id"
    vignettes={Array []}
    visualizations={Array []}
  >
    <div
      style={
        Object {
          "padding": "20px",
        }
      }
    >
      <Grid
        centered={true}
        columns={2}
        padded={true}
        stackable={true}
        stretched={false}
      >
        <div
          className="ui centered stackable padded two column grid"
        >
          <GridColumn>
            <div
              className="column"
            >
              No visualizations!
            </div>
          </GridColumn>
        </div>
      </Grid>
    </div>
  </UnconnectedDynamicsPage>
</Connect(UnconnectedDynamicsPage)>
`;

exports[`DynamicsPage Should match existing snapshot when no props are provided. 1`] = `
<div
  style={
    Object {
      "padding": "20px",
    }
  }
>
  <Grid
    centered={true}
    columns={2}
    padded={true}
    stackable={true}
    stretched={false}
  >
    <GridColumn>
      No visualizations!
    </GridColumn>
  </Grid>
</div>
`;
